pylint/checkers/strings.py

```diff
@@ -892,6 +899,55 @@ def process_non_raw_string_token(
             # character can never be the start of a new backslash escape.
             index += 2
 
+    @check_messages("possible-f-string-as-string")
+    def visit_const(self, node: astroid.Const):
+        self._detect_possible_f_string(node)
+
+    def _detect_possible_f_string(self, node: astroid.Const):
+        """Check whether strings include local/global variables in '{}'
+        Those should probably be f-strings's
+        """
+
+        def detect_if_used_in_format(node: astroid.Const, assign_name: str) -> bool:
+            """Check if the node is used in a call to format() if so return True"""
+            # the skip_class is to make sure we don't go into inner scopes, but might not be needed per se
+            for attr in node.scope().nodes_of_class(
+                astroid.Attribute, skip_klass=(astroid.FunctionDef,)
+            ):
+                if isinstance(attr.expr, astroid.Name):
+                    if attr.expr.name == assign_name and attr.attrname == "format":
+                        return True
+            return False
+
+        if node.pytype() == "builtins.str" and not isinstance(
+            node.parent, astroid.JoinedStr
+        ):
+            # Find all pairs of '{}' within a string
+            inner_matches = re.findall(r"(?<=\{).*?(?=\})", node.value)
+            if len(inner_matches) != len(set(inner_matches)):
+                return
+            if inner_matches:
+                for match in inner_matches:
+                    # Check if match is a local or global variable
+                    if node.scope().locals.get(match) or node.root().locals.get(match):
```

Comment:
```suggestion
                    if not(node.scope().locals.get(match) or node.root().locals.get(match)):
                        return
                    ... 
```
I would do a premature return to avoid the long if / else.